website/client/src/components/snackbars/notifications.vue

```diff
@@ -19,40 +29,294 @@
+
+      this.visibleNotifications.splice(foundNotification, 1);
+
+      this.updateAllowedToFillAgain();
+      this.$store.dispatch('snackbars:remove', $event);
+
+      this.debug('removed', {
+        allowedToFillAgain: this.allowedToFillAgain,
+        storeLength: this.notificationStore.length,
+      });
+
+      if (this.allowedToFillAgain) {
+        // reset the flag so that the next call (for a new notification) will be immediately
+        if (this.visibleNotificationsWithoutErrors.length !== 0) {
+          this.debug('start timeout to fill again');
+          setTimeout(() => {
+            this.debug('before fill new notifications');
+            this.triggerFillUntilFull();
+
+            this.triggerRemovalTimerIfAllowed();
+          }, delayDeleteAndNew);
+        }
+      }
+    },
+    fillVisibleNotifications (notifications) {
+      this.debug({
+        fillAgain: this.allowedToFillAgain,
+        visible: this.visibleNotifications.length,
+        notifications: notifications.length,
+      });
+
+      // the generic checks - new notification array has enough items
+      // is allowed to be filled and don't do anything while the visible items are 2
+      if (notifications.length === 0 || !this.allowedToFillAgain
+          || this.visibleNotifications.length === notificationsVisibleAtOnce) {
+        this.debug('stop fill - 1');
+        return;
+      }
+
+      // this checks if the visible items are the current ones in the notifications array
+      // if so - there is no need to continue, for example on only one visible notification,
+      // it doesn't need to go with the loop again
+      if (notifications.length === this.visibleNotifications.length) {
+        const visibleIds = this.visibleNotifications.map(n => n.uuid);
+
+        const allTheSame = notifications.every(n => visibleIds.includes(n.uuid));
+
+        if (allTheSame) {
+          this.debug('stop fill - 2', {
+            visibleIds,
+            notifications: notifications.length,
+            notificationsStore: this.notificationStore.length,
+          });
+          return;
+        }
+      }
+
+      // fill the new items that needs to be visible
+      if (this.visibleNotifications.length < notificationsVisibleAtOnce) {
+        const visibleIds = this.visibleNotifications.map(n => n.uuid);
+
+        const notAddedYet = notifications.filter(n => !visibleIds.includes(n.uuid));
+
+        this.debug({
+          visibleIds,
+          notAddedYet: notAddedYet.length,
+        });
+
+        if (notAddedYet.length > 0) {
+          this.visibleNotifications.push(notAddedYet[0]);
+        }
+      }
+
+      this.updateAllowedToFillAgain();
+    },
+    async triggerFillUntilFull () {
+      for (let i = 0; i < notificationsVisibleAtOnce; i += 1) {
+        this.debug(`fill ${i}`);
+        this.fillVisibleNotifications(this.notificationStore);
+
+        await sleepAsync(delayFilling); // eslint-disable-line no-await-in-loop
```

Comment:
Could this be done by `map`ping a slice of notificationStore instead of ignoring the relevant eslint rule?

Reply:
Oh, never mind, I misread this.